### PR TITLE
SC-1967- Refactor code to use the new lookup table

### PR DESCRIPTION
**Pull request by aishwa8141**
Moved methods checkExternalIdUniqueness, throwExternalIDNotFoundException, throwUserAlreadyExistsException, from Util.java  to UserLookUp.java, email and phone Uniqueness checks also moved to UserLookUp.java from UserUtil.java

Changed getRecordsByProperties (before it was using allow filtering) -> getRecordsByPropertiesWithFiltering 
Changes now  getRecordsByProperties() does not use allowFiltering 
added   boolean allowFiltering,getRecordsByProperties to reduce duplication of code

removed unused code in SearchHandlerActorTest.java 
removed unused method getUsersByProperties in UserDao.java, UserDaoImpl.java

SomeClass like  OrganisationManagementActor, ShadowUserMigrationScheduler uses getRecordsByProperties()  renamed to  getRecordsByPropertiesWithFiltering()

### CODE REVIEW

File: actors/sunbird-lms-mw/actors/common/src/main/java/org/sunbird/feed/impl/FeedServiceImpl.java [hunk 1]
```diff
@@ -85,7 +85,7 @@ public Response update(Feed feed, RequestContext context) {
     logger.info(context, "FeedServiceImpl:getRecordsByProperties method called : ");
     Response dbResponse =
         getCassandraInstance()
-            .getRecordsByProperties(
+            .getRecordsByPropertiesWithFiltering(
```
Comment (indrajra) [inline actors/sunbird-lms-mw/actors/common/src/main/java/org/sunbird/feed/impl/FeedServiceImpl.java:88]: Check this flow, it maybe that we are querying the feed table with user id. If not, we got to fix this by a script @reshmi-nair , @sknirmalkar89 .
The feeds are more or less an empty table - not many takers - so we needn't even write a script, just drop and recreate with some code adjustments. We can take it separately as a PR.

File: actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-cassandra-utils/src/main/java/org/sunbird/cassandraimpl/CassandraOperationImpl.java [hunk 2]
```diff
@@ -312,6 +324,38 @@ public Response getAllRecords(String keyspaceName, String tableName, RequestCont
     return response;
   }
 
+  @Override
+  public Response getAllRecords(
```
Comment (indrajra) [inline actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-cassandra-utils/src/main/java/org/sunbird/cassandraimpl/CassandraOperationImpl.java:328]: is the existing getAllRecords() function calling this? Asking if this statements are duplicated in another function.

File: service/app/util/AuthenticationHelper.java [hunk 3]
```diff
@@ -77,7 +77,7 @@ public static String verifyClientAccessToken(String clientId, String clientToken
     String validClientId = JsonKey.UNAUTHORIZED;
     try {
       Response clientResponse =
-          cassandraOperation.getRecordsByProperties(
+          cassandraOperation.getRecordsByPropertiesWithFiltering(
```
Comment (indrajra) [inline service/app/util/AuthenticationHelper.java:80]: I think this function itself can be removed. Not actively used - all consumption clients send us a token and we dont have to support this.

**Review by indrajra — CHANGES_REQUESTED** [review]

File: actors/sunbird-lms-mw/actors/common/src/main/java/org/sunbird/feed/impl/FeedServiceImpl.java (hunk 1 above)
Comment (aishwa8141) [reply actors/sunbird-lms-mw/actors/common/src/main/java/org/sunbird/feed/impl/FeedServiceImpl.java:88]: @indrajra  we are querying the feed table with user id and category, the user_feed table primary key is id, and userid and category are indexed

File: actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-cassandra-utils/src/main/java/org/sunbird/cassandraimpl/CassandraOperationImpl.java (hunk 2 above)
Comment (aishwa8141) [reply actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-cassandra-utils/src/main/java/org/sunbird/cassandraimpl/CassandraOperationImpl.java:309]: yes, removed the duplicates, it is calling existing one getAllRecords(),

File: actors/sunbird-lms-mw/actors/common/src/main/java/org/sunbird/feed/impl/FeedServiceImpl.java (hunk 1 above)
Comment (indrajra) [reply actors/sunbird-lms-mw/actors/common/src/main/java/org/sunbird/feed/impl/FeedServiceImpl.java:88]: Yes, I realized so. But we can also fix it.

File: service/app/util/AuthenticationHelper.java (hunk 3 above)
Comment (indrajra) [reply service/app/util/AuthenticationHelper.java:80]: Will await for a better day/PR to get this removed.

**Review by indrajra — APPROVED** [review]

**Pull request merged**